youtube_dl/extractor/tiktok.py

```diff
@@ -94,13 +103,74 @@ class TikTokIE(TikTokBaseIE):
     }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        webpage = self._download_webpage(
-            'https://m.tiktok.com/v/%s.html' % video_id, video_id)
-        data = self._parse_json(self._search_regex(
-            r'\bdata\s*=\s*({.+?})\s*;', webpage, 'data'), video_id)
-        return self._extract_aweme(data)
+        video_id = url.split('/')[-1]
+
+        # extract meta data using the official api
+        # Response json contains: provider url, title, html, author_namee, height, thumbnail_width, width, version,
+        # author_url, thumbnail_height, thumbnail_url, type, provider_name (tiktok)
+
+        json_api = self._download_json('https://www.tiktok.com/oembed?url=' + url, video_id)
+
+        # extract metadata with beautifulSoup
+        webpage = self._download_webpage(url, video_id)
+        soup = BeautifulSoup(webpage, features="html.parser")
+        h2 = soup.find_all("h2", {"class": "jsx-1038045583 jsx-3192540912 jsx-2150087249 video-meta-count"})
```

Comment:
are these class names constant? they look like they are generated with the code (those that looks like jsx-number)
I would have liked to see something more robust than using data[index]. This code is a bit unclear and looks like it can break on small UI changes on tik-tok.